filters/builtin/compress.go

```diff
@@ -20,14 +21,16 @@ const bufferSize = 8192
 
 type encoding struct {
 	name string
-	q    float32
+	q    float32 // encoding client priority
+	p    int     // encoding default priority
 }
 
 type encodings []*encoding
 
 type compress struct {
-	mime  []string
-	level int
+	mime     []string
+	level    int
+	encoding []string
```

Comment:
I suggest to use a map here like
```go
map[string]int {
  "br": 1,
  "gzip": 2,
}
```
this should simplify priority lookup and would not require auxiliary `index` method

Reply:
no that's more complex than having a slice that has an order already

Reply:
it is indeed but as we need to loop through all the client supported encodings to find the "best match", the complexity is probably worth it? (implemented)

Reply:
The suggestion was implemented. Map lookup is cleaner and does not require auxiliary `index` method that iterates slice to find a value.